Documentation/how-to-build-and-run-ilcompiler-in-vscode.md

```diff
@@ -0,0 +1,58 @@
+_Please ensure that [pre-requisites](prerequisites-for-building.md) are installed for a successful build of the repo._
+
+_Note_:
+
+* Instructions below assume ```~/corert``` is the repo root.
+
+# Setting up #
+
+Please make sure you have latest VS Code, C# extension, and .NET Core available. This guide is tested under C# 1.6.2 + VS Code 1.8.1 + CLI 1.0.0-preview4-004233.
+
+This guide assumes that your VS code workspace is set to the root of the repo. 
+
+# Running VS Code
+
+We've checked-in reasonable default ```launch.json``` and ```tasks.json``` under ```corert/.vscode``` directory. You only need to run vscode form corert root:
+
+```
+code ~/corert
+```
+
+And then press SHIFT+COMMAND+B to start the build.
+
+# Debugging ILC.exe using .NET Core Debugger #
+
+Go to the debug pane and click Debug, choose .NET Core as the environment. If needed, you can change program property in launch.json (the gear button) to point to a different flavor of ilc:
+
+```json
+            "osx": {
+                "program": "${workspaceRoot}/bin/Product/OSX.x64.Debug/packaging/publish1/ilc.dll"
+            },
+            "linux": {
+                "program": "${workspaceRoot}/bin/Product/Linux.x64.Debug/packaging/publish1/ilc.dll"
+            },
+```
+
+By default we've disabled automatic build before debug. If you want to change that, you can change the ```preLaunchTask``` property to ```"build"```. But this is not currently recommended.
+
+The magic to make this work are two changes (automated in the building/packaging process):
```

Comment:
I would add description of these changes to the relevant lines in the .csproj file instead. It is not something that somebody going through these steps needs to know.